sympy/utilities/iterables.py

```diff
@@ -322,7 +327,37 @@ def multiset(seq: Sequence[T]) -> dict[T, int]:
 
 
 
-def ibin(n, bits=None, str=False):
+@overload
+def ibin(
+    n: SupportsIndex,
+    bits: SupportsIndex | None = None,
+    str: Literal[False] = False,
+) -> list[int]: ...
+@overload
+def ibin(
+    n: SupportsIndex,
+    bits: SupportsIndex | None = None,
+    str: Literal[True] = True,
+) -> str: ...
```

Comment:
The `= True` default value should be removed here. If you look at this with pyright you can see the problem:
<img width="1055" height="346" alt="Screenshot 2025-12-24 at 16 18 05" src="https://github.com/user-attachments/assets/12824a7e-ecf8-4441-9d1c-6141e8903d68" />
I think it should be like:
```
@overload
def ibin(
    n: SupportsIndex,
    bits: SupportsIndex | None = None,
    *,
    str: Literal[True],
) -> str: ...
```
with `*` as the keyword argument separator added to all of the overloads and the main signature.